src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +752,88 @@ const UsersConfigurationRow = ({
     );
 };
 
+// This method needs to be outside of component as re-render would create a new instance of debounce
+const validateKey = debounce(500, (key, setKeyObject) => {
+    if (isEmpty(key))
+        return;
+
+    return cockpit.script(`echo ${key} | ssh-keygen -l -f /dev/stdin`, { error: "message" })
+            .then(() => {
+                const parts = key.split(" ");
+                if (parts.length > 2) {
+                    setKeyObject({
+                        type: parts[0],
+                        data: parts[1],
+                        comment: parts[2], // comment is optional in SSH-format
+                    });
+                }
+            })
+            .catch(() => {
+                setKeyObject(undefined);
+                console.warn("Could not validate a public key");
+            });
+});
+
+const SshKeysRow = ({
+    id, item, onChange, idx, removeitem,
+}) => {
+    const [keyObject, setKeyObject] = useState();
+
+    return (
+        <Grid id={id} key={id}>
+            <GridItem span={11}>
+                <FormGroup label={_("Public key")} fieldId={id + "-public-key"}>
+                    {keyObject
+                        ? <FlexItem id="validated">
+                            <strong>{keyObject.comment}</strong>
+                            <span>{keyObject.comment ? " - " + keyObject.type : keyObject.type}</span>
+                            <div>{keyObject.data}</div>
+                        </FlexItem>
+                        : <TextArea value={item.value || ""}
+                            aria-label={_("Public SSH key")}
+                            onChange={(_, value) => {
+                                onChange(idx, "value", value);
+                                validateKey(value, setKeyObject);
```

Comment:
These 15 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20231003-232224-a08f566d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)